plugins/TelescopeControl/src/TelescopeClient.cpp

```diff
@@ -125,6 +126,31 @@ TelescopeClient *TelescopeClient::create(const QString &url)
 TelescopeClient::TelescopeClient(const QString &name) : nameI18n(name), name(name)
 {}
 
+
+bool TelescopeClient::GetAltAzFromJ2000Position(const Vec3d &j2000Pos,TelescopeControl::Equinox equinox ,double& alt, double &az) const
+{
+   const StelCore* core = StelApp::getInstance().getCore();
+   const  bool res = core != nullptr;
+   if(res)
+   {
+      const bool equinoxJNow = equinox == TelescopeControl::EquinoxJNow;
+      const StelCore::RefractionMode mode = equinoxJNow ? StelCore::RefractionOff : StelCore::RefractionAuto;
```

Comment:
The necessity for refraction correction should depend on atmosphere visibility. It has nothing to do with JNow setting. What is your motivation to connect them?

Reply:
Actually I'm not sure what it is used for. The reason I have this check is because I saw in the TelescopeGoTo method that it was used to correct the J2000 position. See line 300-304:
![image](https://github.com/Stellarium/stellarium/assets/16615085/e38599de-c650-4ed3-af5a-b98f33d02eba)

Is there a way to check for atmosphere visilibility. And if so should it also be used in the screenshot above?

Reply:
IMO, when driving a telescope from a terrestrial location, it is safe to assume (the real, physical) atmosphere (in which we breathe) is on. 

If you have "pure" J2000 data from a catalog/ephemeris (without refraction correction), You must convert them to JNow equatorial position without refraction correction (core->j2000ToEquinoxEqu(j2000Pos, RefractionOff) when the telescope expects that (and applies refraction correction internally). 


Then, to compute geometrical Alt-Az position for a goto capable device, you compute alt-az position  (core->j2000ToAltAz(positionJ2000, refractionMode);)  If your telescope expects AltAz coordinates to be corrected for refraction, you should compute with RefractionOn. If your telescope expects AltAz position to be without refraction correction (because it will do that by itself!), you use RefractionOff. 

I have never used an alt-az GOTO telescope, so I am not familiar with these protocols. Maybe this needs yet another config option ("Goto is AzAlt and expects/does not expect refraction correction") Does such a device track the object after the goto slew? Why not just send RA/Dec to a properly configured/set up mount?

Reply:
I know that my own Eq6-r pro does refraction correction based on the elevation you input. I don't know about other mounts.
Seeing that the class I'm extending is the TelescopeTCP class which is most likely used for (I asumme??) for a homemade telescope control system maybe it would be wise to have these options under the Telescope properties when you set up the host.

For example:
<img width="268" alt="image" src="https://github.com/Stellarium/stellarium/assets/16615085/1ea83093-8425-442f-a6d7-e4ce06ecd9fa">

What do you think?

Reply:
Probably the Boolean JNow setting will have to be extended to a 3-value J2000/JNow/AltAz, and an additional Boolean RefractionCorrection would govern the independent extra. @alex-w ?

Reply:
@huntflex what is the final target for this patch?

Reply:
@alex-w  I wanted to extend the relevant telescope classes with altitude azimuth. I thought a natural (and easier place) to begin with would be the telescopeTCP class. As you know the TelescopeTCP class is the one instantiated whenever a user configures a telescope with a "remote computer". I assumed that one would also want access to alt/az in this kind of setup since a "remote computer" might not be limited to an equatorial mount.

Reply:
Well, the mount can be AltAz-only and I fear these mounts will not understand some parts of protocol. Plus EQ/AltAz-mount requires setting the mode. In both cases changes of protocols can be for first item only - commnication directly through a serial port. The second item is incompatible now (I really don't know modern use cases for old Stellarium protocol for mounts - a TelescopeServer).

What about real hardware - is it working in AltAz mode?

Reply:
This may be useful for DIY projects with self-made controllers (e.g. Arduino, self-made axis decoder/motor protocol, ...). But I have never used the telescopeTCP stuff, so I cannot say how relevant this still is. 
The docs in TelescopeClient.hpp indicate a "Stellarium telescope control protocol" to be included in "telescope server software". Where is this all? (Yes, Google found a 1-page file from J.G. on some "free" site.) Is/Was that a reference implementation of something useful? Then it should probably be archived in a repo here. And somebody who was involved should write the design docs to explain it all. (OK, dreaming...)

Reply:
The first version of TelescopeServer is located in SourceForge (SVN: https://sourceforge.net/p/stellarium/code/HEAD/tree/trunk/), the second version is in GitHub: https://github.com/johannesgajdosik/TelescopeServer2

Reply:
So I'm wondering if this is the right course of action with this patch. Whether we should continue with this PR?